flang/include/flang/Optimizer/Dialect/FIROps.td

```diff
@@ -3074,6 +3074,7 @@ def fir_DeclareOp : fir_Op<"declare", [AttrSizedOperandSegments,
     AnyRefOrBox:$memref,
     Optional<AnyShapeOrShiftType>:$shape,
     Variadic<AnyIntegerType>:$typeparams,
+    Optional<I1>:$dummy_scope,
```

Comment:
Is there a reason it's an I1 type? Could it be its own type like it's done for the async token for example.

Reply:
I do not have a strong preference for the result type, so I picked an existing one.  A dedicated type is probably more robust, e.g. if some code tries to use the dummy-scope op's result in arithmetic it will be an error.  Do you think a dedicated type will be better?

Reply:
I don't have strong opinion too. If it's not too complicated to have a dedicated type I think it will make it nicer but I can live with the I1 too.